gato/training/trainer.py

```diff
@@ -105,9 +108,13 @@ def train_iteration(self, num_steps, iter):
     def train_step(self):
         logs = {}
         logs['training/learning_rate'] = self.scheduler.get_lr()[0] # store LR at current step
-        
+
         # Build training batch
-        batch_dicts = self.sample_control_batch(self.args.batch_size)
+        # for simplicity sake - assume interleave training on control & text
+        if self.steps % 2 == 0:
+            batch_dicts = self.sample_control_batch(self.args.batch_size)
+        else:
+            batch_dicts = self.process_text_batch(next(iter(self.text_dataset)))
```

Comment:
the format of the dictionary currently returned by your process_text_batch is not compatible with input to forward() -> tokenize_input_dicts, as it expects each sequence to be a separate dictionary in a list. Some discussion on reworking this can be found here: https://github.com/ManifoldRG/gato-control/issues/5

Reply:
Resolved in latest commit.